hack/check_root.sh

```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+if ! [ $(id -u) = 0 ]; then
```

Comment:
It took me a long while to see the negation; this might be a little more readable
```bash
if [ "$(id -u)" != "0" ]; then
```

Reply:
Done, thanks for checking!

Reply:
I just found yesterday that bats sets an environment varible EUID, which means you would not need to execute a program

if [ "${EUID}" != "0" ]; then

Reply:
(bash, not bats). But yes, that's a good point. I've always been a little reluctant to trust shell-defined variables (`$HOME`, `$LOGNAME`) but this one is explicitly documented as readonly in bash(1) so I think it's ok. For future reference, since this just merged.